valhalla/midgard/sequence_writer.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
```

Comment:
instead of having yet another header can you move this into the existing sequence header?